api/repository.go

```diff
@@ -57,15 +59,16 @@ type RepositoryCreate struct {
 	ProjectID int
 
 	// Domain specific fields
-	Name               string `jsonapi:"attr,name"`
-	FullPath           string `jsonapi:"attr,fullPath"`
-	WebURL             string `jsonapi:"attr,webUrl"`
-	BranchFilter       string `jsonapi:"attr,branchFilter"`
-	BaseDirectory      string `jsonapi:"attr,baseDirectory"`
-	FilePathTemplate   string `jsonapi:"attr,filePathTemplate"`
-	SchemaPathTemplate string `jsonapi:"attr,schemaPathTemplate"`
-	SheetPathTemplate  string `jsonapi:"attr,sheetPathTemplate"`
-	ExternalID         string `jsonapi:"attr,externalId"`
+	Name                string                     `jsonapi:"attr,name"`
+	FullPath            string                     `jsonapi:"attr,fullPath"`
+	WebURL              string                     `jsonapi:"attr,webUrl"`
+	BranchFilter        string                     `jsonapi:"attr,branchFilter"`
+	BaseDirectory       string                     `jsonapi:"attr,baseDirectory"`
+	FilePathTemplate    string                     `jsonapi:"attr,filePathTemplate"`
+	SchemaMigrationType ProjectSchemaMigrationType `jsonapi:"attr,schemaMigrationType"`
```

Comment:
schemaMigrationType should only be part of project entity. Repository entity doesn't need it.

Reply:
We need it for now for the frontend to work properly with minimal change.

Reply:
We can split the change and get help from @boojack to make Frontend happy.

Reply:
Based on my rudimentary frontend skills, I think it could be quite a bit of refactoring involved in order to make CRUD project settings directly (vs. proxy changes through `RepositoryCreate` and `RepositoryPatch`) on the "Version control" page, we also only re-fetch `repository` after saving the changes on this page.

@boojack Please take a look and maybe there is an easy path to solve this problem 🙇 

@d-bytebase - Not sure what you mean exactly by "split the change", by commits in this same PR or address this UI issue in another PR by @boojack (in which case remove this field from the `Repository` struct and leave the UI in a broken state in dev)?

Reply:
I mean making the 1st PR for the ideal backend change.

Reply:
@d-bytebase - I extracted backend changes in https://github.com/bytebase/bytebase/pull/2333, PTAL.

@boojack - I'm leaving this PR as a reference implementation, so you know what relevant frontend changes should be, LMK if you have any questions!